x/crosschain/keeper/msg_server_whitelist_erc20.go

```diff
@@ -0,0 +1,122 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	tmbytes "github.com/tendermint/tendermint/libs/bytes"
+	tmtypes "github.com/tendermint/tendermint/types"
+	"github.com/zeta-chain/zetacore/common"
+	"github.com/zeta-chain/zetacore/x/crosschain/types"
+	fungibletypes "github.com/zeta-chain/zetacore/x/fungible/types"
+	zetaObserverTypes "github.com/zeta-chain/zetacore/x/observer/types"
+)
+
+func (k Keeper) WhitelistERC20(goCtx context.Context, msg *types.MsgWhitelistERC20) (*types.MsgWhitelistERC20Response, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	if msg.Creator != k.zetaObserverKeeper.GetParams(ctx).GetAdminPolicyAccount(zetaObserverTypes.Policy_Type_deploy_fungible_coin) {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrUnauthorized, "Deploy can only be executed by the correct policy account")
+	}
+	erc20Addr := ethcommon.HexToAddress(msg.Erc20Address)
+	if erc20Addr == (ethcommon.Address{}) {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid ERC20 contract address (%s)", msg.Erc20Address)
+	}
+
+	// check if it's already whitelisted
+	foreignCoins := k.fungibleKeeper.GetAllForeignCoins(ctx)
+	for _, fcoin := range foreignCoins {
+		assetAddr := ethcommon.HexToAddress(fcoin.Asset)
+		if assetAddr == erc20Addr && fcoin.ForeignChainId == msg.ChainId {
+			return nil, sdkerrors.Wrapf(types.ErrInvalidAddress, "ERC20 contract address (%s) already whitelisted on chain (%d)", msg.Erc20Address, msg.ChainId)
+		}
+	}
+	// TODO: check if this chain is supported
+	chain := common.GetChainFromChainID(msg.ChainId)
```

Comment:
What does it return if the chain is not supported?
IMO It seems to me an important error to handle to avoid unexpected changes

Reply:
Actually, the actual question is what is returned if there is no chain associated with chain ID?
If it is `nil` we should handle this to avoid a `panic`

Reply:
sorry missed this one. 
Now it checks nil and also whether this chain is supported or not.